<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsMeshRendererVectorSettingsWidgetBase</class>
 <widget class="QWidget" name="QgsMeshRendererVectorSettingsWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>426</width>
    <height>1006</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="16" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="7" column="0" colspan="2">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0" rowspan="2">
      <widget class="QLabel" name="filterByMagnitudeLabel">
       <property name="text">
        <string>Filter by magnitude</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="maximumMagLabel">
       <property name="text">
        <string>Max</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QgsDoubleSpinBox" name="mMaxMagSpinBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimum">
        <double>-1000000000000000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000000000000000.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="minimumMagLabel">
       <property name="text">
        <string>Min</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QgsDoubleSpinBox" name="mMinMagSpinBox">
       <property name="minimum">
        <double>-1000000000000000.000000000000000</double>
       </property>
       <property name="maximum">
        <double>1000000000000000.000000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="5" column="1">
    <widget class="QgsColorButton" name="mColorWidget">
     <property name="minimumSize">
      <size>
       <width>120</width>
       <height>0</height>
      </size>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QgsCollapsibleGroupBox" name="mColorRampShaderGroupBox">
     <property name="title">
      <string>Color Ramp Shader</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="2">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Max</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="mColorRampShaderLoadButton">
        <property name="toolTip">
         <string>Load</string>
        </property>
        <property name="icon">
         <iconset resource="../../../images/images.qrc">
          <normaloff>:/images/themes/default/mActionRefresh.svg</normaloff>:/images/themes/default/mActionRefresh.svg</iconset>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="5">
       <widget class="QgsColorRampShaderWidget" name="mColorRampShaderWidget" native="true">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="5">
       <widget class="Line" name="line">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QgsDoubleSpinBox" name="mColorRampShaderMaximumSpinBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <double>-1000000000000000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000000000000000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Min</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsDoubleSpinBox" name="mColorRampShaderMinimumSpinBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <double>-1000000000000000.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1000000000000000.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QComboBox" name="mColoringMethodComboBox"/>
   </item>
   <item row="3" column="1">
    <widget class="QgsDoubleSpinBox" name="mLineWidthSpinBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QComboBox" name="mSymbologyVectorComboBox">
     <item>
      <property name="text">
       <string>Arrows</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Streamlines</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Traces</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Wind Barbs</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QWidget" name="widget" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
     </layout>
    </widget>
   </item>
   <item row="11" column="0" colspan="2">
    <widget class="QGroupBox" name="mHeadOptionsGroupBox">
     <property name="title">
      <string>Head Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="1" column="2">
       <widget class="QLabel" name="percShaftLenLabel_2">
        <property name="text">
         <string>% of shaft length</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsDoubleSpinBox" name="mHeadWidthSpinBox">
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QgsDoubleSpinBox" name="mHeadLengthSpinBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="percShaftLenLabel">
        <property name="text">
         <string>% of shaft length</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="headLengthLabel">
        <property name="text">
         <string>Length</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="headWidthLabel">
        <property name="text">
         <string>Width</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>Coloring Method</string>
     </property>
    </widget>
   </item>
   <item row="13" column="0" colspan="2">
    <widget class="QGroupBox" name="mStreamlineWidget">
     <property name="title">
      <string>Streamlines Seeding Method</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="3,2">
      <item>
       <widget class="QComboBox" name="mStreamlinesSeedingMethodComboBox">
        <item>
         <property name="text">
          <string>On Mesh/Grid</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Randomly</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="mStreamlinesDensityLabel">
          <property name="text">
           <string>Density</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QgsDoubleSpinBox" name="mStreamlinesDensitySpinBox">
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="value">
           <double>5.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="10" column="0" colspan="2">
    <widget class="QGroupBox" name="mDisplayVectorsOnGridGroupBox">
     <property name="title">
      <string>Display on User Grid</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="xSpacingLabel">
        <property name="text">
         <string>X Spacing</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsSpinBox" name="mXSpacingSpinBox">
        <property name="suffix">
         <string> px</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>8000</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="ySpacingLabel">
        <property name="text">
         <string>Y Spacing</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QgsSpinBox" name="mYSpacingSpinBox">
        <property name="suffix">
         <string> px</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>5000</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="symbologyLabel">
     <property name="text">
      <string>Symbology</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Line width</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="mSingleColorLabel">
     <property name="text">
      <string>Color</string>
     </property>
    </widget>
   </item>
   <item row="12" column="0" colspan="2">
    <widget class="QGroupBox" name="mArrowLengthGroupBox">
     <property name="title">
      <string>Arrow Length</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QComboBox" name="mShaftLengthComboBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="sizeAdjustPolicy">
         <enum>QComboBox::AdjustToContents</enum>
        </property>
        <item>
         <property name="text">
          <string>Defined by Min and Max</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Scaled to Magnitude</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Fixed</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="mShaftOptionsStackedWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="page">
         <layout class="QFormLayout" name="formLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="2" column="0">
           <widget class="QLabel" name="minimumShaftLabel">
            <property name="text">
             <string>Minimum</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="maximumShaftLabel">
            <property name="text">
             <string>Maximum</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QgsDoubleSpinBox" name="mMinimumShaftSpinBox"/>
          </item>
          <item row="4" column="1">
           <widget class="QgsDoubleSpinBox" name="mMaximumShaftSpinBox"/>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QFormLayout" name="formLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="1">
           <widget class="QgsDoubleSpinBox" name="mScaleShaftByFactorOfSpinBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="scaleByFactorOfLabel">
            <property name="text">
             <string>Scale by a factor of</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_3">
         <layout class="QFormLayout" name="formLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="1">
           <widget class="QgsDoubleSpinBox" name="mShaftLengthSpinBox"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="LengthLabel">
            <property name="text">
             <string>Length</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="14" column="0" colspan="2">
    <widget class="QGroupBox" name="mTracesGroupBox">
     <property name="title">
      <string>Traces</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Particles count</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QgsSpinBox" name="mTracesParticlesCountSpinBox">
          <property name="maximum">
           <number>1000000</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="value">
           <number>1000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <property name="topMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Max tail length</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QgsDoubleSpinBox" name="mTracesMaxLengthSpinBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>99999999999999.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>10.000000000000000</double>
          </property>
          <property name="value">
           <double>10.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QgsUnitSelectionWidget" name="mTracesTailLengthMapUnitWidget" native="true"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="15" column="0" colspan="2">
    <widget class="QGroupBox" name="mWindBarbGroupBox">
     <property name="title">
      <string>Wind Barbs</string>
     </property>
     <layout class="QFormLayout" name="formLayout_4">
      <item row="1" column="0">
       <widget class="QLabel" name="windBarbLengthLabel">
        <property name="text">
         <string>Length</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="windBarbUnitsLabel">
        <property name="text">
         <string>Data units</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="mWindBarbUnitsComboBox">
        <property name="toolTip">
         <string>Select the units the data are in.&lt;br&gt;Values are converted to knots for rendering the wind barbs.</string>
        </property>
        <item>
         <property name="text">
          <string>m/s</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>km/h</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>knots</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>mi/h</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>ft/s</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>other units</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="mWindBarbMagnitudeMultiplierLabel">
        <property name="text">
         <string>Multiplier</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QgsDoubleSpinBox" name="mWindBarbMagnitudeMultiplierSpinBox">
        <property name="toolTip">
         <string>Data will be multiplied by this value to be converted to knots (nautical miles per hour)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QgsDoubleSpinBox" name="mWindBarbLengthSpinBox">
          <property name="toolTip">
           <string>This defines the shaft length.&lt;br&gt;The pennants and barbs are scaled proportionally.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QgsUnitSelectionWidget" name="mWindBarbLengthMapUnitWidget" native="true"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgsdoublespinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsUnitSelectionWidget</class>
   <extends>QWidget</extends>
   <header>qgsunitselectionwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsColorRampShaderWidget</class>
   <extends>QWidget</extends>
   <header>qgscolorrampshaderwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsColorButton</class>
   <extends>QToolButton</extends>
   <header>qgscolorbutton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mSymbologyVectorComboBox</tabstop>
  <tabstop>mLineWidthSpinBox</tabstop>
  <tabstop>mColoringMethodComboBox</tabstop>
  <tabstop>mColorWidget</tabstop>
  <tabstop>mColorRampShaderMinimumSpinBox</tabstop>
  <tabstop>mColorRampShaderMaximumSpinBox</tabstop>
  <tabstop>mColorRampShaderLoadButton</tabstop>
  <tabstop>mMinMagSpinBox</tabstop>
  <tabstop>mMaxMagSpinBox</tabstop>
  <tabstop>mDisplayVectorsOnGridGroupBox</tabstop>
  <tabstop>mXSpacingSpinBox</tabstop>
  <tabstop>mYSpacingSpinBox</tabstop>
  <tabstop>mHeadWidthSpinBox</tabstop>
  <tabstop>mHeadLengthSpinBox</tabstop>
  <tabstop>mShaftLengthComboBox</tabstop>
  <tabstop>mMinimumShaftSpinBox</tabstop>
  <tabstop>mMaximumShaftSpinBox</tabstop>
  <tabstop>mStreamlinesSeedingMethodComboBox</tabstop>
  <tabstop>mStreamlinesDensitySpinBox</tabstop>
  <tabstop>mTracesParticlesCountSpinBox</tabstop>
  <tabstop>mTracesMaxLengthSpinBox</tabstop>
  <tabstop>mScaleShaftByFactorOfSpinBox</tabstop>
  <tabstop>mShaftLengthSpinBox</tabstop>
 </tabstops>
 <resources>
  <include location="../../../images/images.qrc"/>
  <include location="../../../images/images.qrc"/>
 </resources>
 <connections/>
</ui>
